### PR TITLE
GFORMS-791: add noneChoiceError

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/validation/ComponentValidator.scala [hunk 1]
```diff
@@ -554,6 +554,23 @@ object ComponentValidator {
...
 
+  def validateChoiceNoneError[D <: DataOrigin](
+    fieldValue: FormComponent,
+    noneChoice: Int,
+    error: LocalisedString
+  )(
+    formModelVisibilityOptics: FormModelVisibilityOptics[D]
+  )(implicit l: LangADT): ValidatedType[Unit] = {
+    val choiceValue = formModelVisibilityOptics.data
+      .get(fieldValue.modelComponentId)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/validation/ComponentValidator.scala:565]: Please call `many` not `get`. 

`many` is for components which can produce multiple answers, which in our case is only choice

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/validation/ComponentValidator.scala (hunk 1 above)
Comment (lxol) [reply app/uk/gov/hmrc/gform/validation/ComponentValidator.scala]: ok, I got inspiration from 
https://github.com/hmrc/gform-frontend/blob/e7d469c09104c79f3039b75270ff3affe5fe6809/app/uk/gov/hmrc/gform/validation/ComponentValidator.scala#L548
should we update it here as well?

File: app/uk/gov/hmrc/gform/validation/ComponentValidator.scala [hunk 2]
```diff
@@ -554,6 +554,23 @@ object ComponentValidator {
...
+    fieldValue: FormComponent,
+    noneChoice: Int,
+    error: LocalisedString
+  )(
+    formModelVisibilityOptics: FormModelVisibilityOptics[D]
+  )(implicit l: LangADT): ValidatedType[Unit] = {
+    val choiceValue = formModelVisibilityOptics.data
+      .many(fieldValue.modelComponentId)
+      .toSeq
+      .flatMap(_.toSeq)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/validation/ComponentValidator.scala:567]:
```suggestion
      .flatten
```
One suggestion

File: app/uk/gov/hmrc/gform/validation/ComponentValidator.scala (hunk 1 above)
Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/validation/ComponentValidator.scala]: Yes, we should change it there as well.

File: app/uk/gov/hmrc/gform/validation/ComponentValidator.scala (hunk 2 above)
Comment (lxol) [reply app/uk/gov/hmrc/gform/validation/ComponentValidator.scala]: done

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**